AdvLoggerPkg/Library/AdvancedLoggerLib/Pei/AdvancedLoggerLib.c

```diff
@@ -46,6 +46,13 @@ AdvancedLoggerWrite (
     ASSERT (AdvancedLoggerPpi->Signature == ADVANCED_LOGGER_PPI_SIGNATURE);
     ASSERT (AdvancedLoggerPpi->Version == ADVANCED_LOGGER_PPI_VERSION);
 
+    if ((AdvancedLoggerPpi->Signature != ADVANCED_LOGGER_PPI_SIGNATURE) ||
+        (AdvancedLoggerPpi->Version != ADVANCED_LOGGER_PPI_VERSION))
+    {
+      // Signature/Version mismatch, don't attempt to use the PPI.

```

Comment:
We can move the asserts inside the if statement here so we don't check the condition twice except on the failure case.